### DISCUSSION TITLE
Unfilled Grain Boundary Issue

### DISCUSSION THREAD

**Discussion by elegantEmily** [mtex-toolbox/mtex#1103] [Ask Anything]
Currently I am using the fill(ebsd, grains) to correct my EBSD data. However, I noticed on some of my datasets the fill function leaves large areas at the grain boundary unfilled (see white areas that grain boundaries outline).

Is there any way to get the grains to grow into this large grain boundary area?

I have also tried to fill(ebsd) to see if it would fill in these areas and I haven't had any luck. When I use calGrains it doesn't draw the grain boundary between the grains, but rather outlines each grain. 

I have attached the unfilled, filled, and filled with grain boundaries for reference.

Thanks and have a wonderful day!
Emily
![N216F_Area1_unfilled](https://user-images.githubusercontent.com/60235316/122964780-3af4db00-d34d-11eb-8fb9-74a649fc360f.png)
![N216F_Area1_filled](https://user-images.githubusercontent.com/60235316/122964796-3d573500-d34d-11eb-99f1-e834f4f1add4.png)
![N216F_Area1_filledgb](https://user-images.githubusercontent.com/60235316/122964807-3fb98f00-d34d-11eb-99c7-d591e4105aee.png)

**Comment by kilir** [comment]
Hi,
fill will only add points in case they are non-existent - that is different to 'notIndexed'. Do the pixels decorating the grain boundary belong to the 'notIndexed' phase?

Also, have a look here: https://mtex-toolbox.github.io/GrainReconstructionAdvanced.html

Cheers,
Rüdiger

**Reply by elegantEmily** [reply]
They are showing up as 'notIndexed' I believe. If I remove these 'notIndexed' pixels from my dataset, will the fill function then fill these points?

Thanks!
Emily

**Reply by elegantEmily** [reply]
I removed the not Indexed points and used the fill function and it worked!

Thanks for your help!
Emily
